stacks-signer/src/signer.rs

```diff
@@ -154,6 +154,42 @@ pub struct Signer {
     pub coordinator_selector: CoordinatorSelector,
     /// The approved key registered to the contract
     pub approved_aggregate_public_key: Option<Point>,
+    /// This is the signer's current view of the burn chain
+    pub current_burn_block_info: Option<BurnBlockInfo>,
+    /// The current active miner's key (if we know it!)
+    pub miner_key: Option<PublicKey>,
+}
+
+/// The signer's view of the current burn block
+pub struct BurnBlockInfo {
+    /// The burn block height
+    pub burn_block_height: u32,
+    /// The active miner
+    pub active_miner: PublicKey,
+    /// The reward cycle
+    pub reward_cycle: u64,
+}
+
+impl Signer {
+    /// Return the current coordinator. If in the active reward cycle, this is the miner,
+    ///  so the first element of the tuple will be None (because the miner does not have a signer index).
```

Comment:
Does this mean DKG is still coordinated by the signers?

Reply:
Yeah, exactly. DKG will still need to be coordinated among signers: miners don't have the same relationship to DKG that they do to block signing (that is: DKG can progress without the current miner. block production _cannot_ progress without the miner).